storage_controller/src/service.rs

```diff
@@ -478,6 +478,7 @@ impl From<DatabaseError> for ApiError {
             DatabaseError::Logical(reason) | DatabaseError::Migration(reason) => {
                 ApiError::InternalServerError(anyhow::anyhow!(reason))
             }
+            DatabaseError::Client(reason) => ApiError::Conflict(reason),
```

Comment:
I'm not sure this always makes sense. Instead of this generic conversion, let's match the DatabaseError in Service::node_register and convert it to an API error inline there.